src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToLocalizedStringTransformer.php

```diff
@@ -130,7 +130,10 @@ public function reverseTransform($value)
         try {
             if ($dateOnly) {
                 // we only care about year-month-date, which has been delivered as a timestamp pointing to UTC midnight
-                return new \DateTime(gmdate('Y-m-d', $timestamp), new \DateTimeZone($this->inputTimezone));
+                $dateTime = new \DateTime(gmdate('Y-m-d', $timestamp), new \DateTimeZone($this->outputTimezone));
+                $dateTime->setTimezone(new \DateTimeZone($this->inputTimezone));
```

Comment:
Yes these are the lines I'm talking about, but I don't get it what do you mean?

I'm not saying this is not correct here, but it may not be needed to set again the timezone, hence the check is missing.